clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -752,6 +758,106 @@ private StoragePartitionConfig getStoragePartitionConfig(
+            break; // only interested in this header here
+          }
+        }
+        if (!isLeaderCompleteHeaderFound) {
+          // reset LeaderCompleteState: If a leader originally sent this header but later is rolled back to a version
+          // that doesn't support this header or this TP gets a new leader which doesn't support this header yet.
+          newState = LEADER_COMPLETE_STATE_UNKNOWN;
+        }
+
+        if (oldState != newState) {
+          LOGGER.info(
+              "LeaderCompleteState for store {} version {} partition {} changed from {} to {}",
+              storeName,
+              versionNumber,
+              partitionConsumptionState.getPartition(),
+              oldState,
+              newState);
+          partitionConsumptionState.setLeaderCompleteState(newState);
+        } else {
+          LOGGER.debug(
+              "LeaderCompleteState for store {} version {} partition {} received from leader: {} and is unchanged from the previous state",
+              storeName,
+              versionNumber,
+              partitionConsumptionState.getPartition(),
+              newState);
+        }
+        if (!partitionConsumptionState.isFirstHeartBeatSOSReceived()) {
+          partitionConsumptionState.setFirstHeartBeatSOSReceived(true);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks whether the lag is acceptable for hybrid stores
+   *
+   * If the instance is a standby or DaVinciClient: Also check if
+   * 1. leaderCompleteStatus header has been received and
+   * 2. leader was completed and
+   * 3. the last update time was within 5 seconds
+   */
+  protected boolean isLagAcceptableForHybridStore(PartitionConsumptionState pcs, long offsetLag, long offsetThreshold) {
+    boolean isLagAcceptable = (offsetLag <= offsetThreshold);
+
+    if (isHybridMode() && (isDaVinciClient || pcs.getLeaderFollowerState().equals(STANDBY))) {
+      if (!isLagAcceptable) {
+        return false;
+      }
+
+      // non AA stores have issues reading HB SOS from the RT leading to the standby replicas waiting for
+      // leader completion state header indefinitely, so disabling it until that issue is resolved
+      if (!(getServerConfig().isLeaderCompleteStateCheckEnabled() && this.isActiveActiveReplicationEnabled())) {
+        return true;
+      }
+
+      // wait for the first HB to know if the leader supports sending LeaderCompleteState or not
+      if (!pcs.isFirstHeartBeatSOSReceived()) {
+        return false;
+      }
+
+      // if the leader don't support LeaderCompleteState
+      if (pcs.getLeaderCompleteState().equals(LEADER_COMPLETE_STATE_UNKNOWN)) {
+        return true;
+      }
+
+      // check if the leader is completed and the last update time was within the configured time
+      return (pcs.isLeaderCompleted()
+          && ((System.currentTimeMillis() - pcs.getLastLeaderCompleteStateUpdateInMs()) <= getServerConfig()
+              .getLeaderCompleteStateCheckValidIntervalMs()));
```

Comment:
1 minute is too strict IMHO. I would start with something around 10 minutes.

Reply:
SG! Could you please add a log here with redundant log filtering: if HB with header is found but not valid anymore, log the value of "System.currentTimeMillis() - pcs.getLastLeaderCompleteStateUpdateInMs()".

Reply:
its already getting logged inside `checkAndLogIfLagIsAcceptableForHybridStore`